pkg/controller/tiers/tiers_controller.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tiers
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/networkpolicy"
+	"github.com/tigera/operator/pkg/render/tiers"
+	"k8s.io/apimachinery/pkg/types"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/controller/options"
+	"github.com/tigera/operator/pkg/controller/status"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// The Tiers controller reconciles system tiers and network policies using the v3 API. v3 resources require the API server to be available.
+// System tiers are reconciled here instead of the installation controller since the API server is not available until after installation completes.
+// Similarly, policies for installation and API server controllers are reconciled here to ensure the API server is available when they are reconciled.
```

Comment:
During further testing, I realized there are more dependency ordering considerations when it comes to managed clusters, since the license key needs to be pushed from the management cluster.

I've added a commit that addresses this. You can check [this comment ](https://github.com/tigera/operator/pull/1970/commits/e070915949aea39d8a358da6ea69b4c53a104461#diff-43340c178ffd2e84ca3fc589d654e19ed933e66852fca37c609c063cc207ea35R46)for an explanation/preview of the change